src/components/ScholarshipForm.jsx

```diff
@@ -137,9 +149,14 @@ function ScholarshipForm({ scholarship, submitFn, onSubmitError }) {
                   color="primary"
                   disabled={formik.isSubmitting}
                   type={onLastStep ? 'submit' : 'button'}
-                  onClick={() =>
-                    !onLastStep && setActiveStep((prevStep) => prevStep + 1)
-                  }>
+                  onClick={() => {
+                    if (onLastStep) return;
+                    formik.validateForm().then((errors) => {
```

Comment:
yes, there is something weird going on there. It might have to do with requiring the fields to be touched before showing the errors. Since it's going to require me going into different components to try and fix it, I think it'd be best to fix this in another PR?

Reply:
If it's non-trivial then yeah a different PR sounds good. You don't want it to block what otherwise looks like a ready PR.